test/component2_suggestors/class_name_and_annotation_migrator_test.dart

```diff
@@ -0,0 +1,205 @@
+
+    test('annotation with args and extending class updates', () {
+      testSuggestor(
+        expectedPatchCount: 2,
+        input: '''
+          @Component(isWrapper: true)
+          class FooComponent extends UiComponent<FooProps> {}
+        ''',
+        expectedOutput: '''
+          @Component2(isWrapper: true)
+          class FooComponent extends UiComponent2<FooProps> {}
+        ''',
+      );
+    });
+
+    test('annotation with args and extending stateful class updates', () {
+      testSuggestor(
+        expectedPatchCount: 2,
+        input: '''
+          @Component(isWrapper: true)
+          class FooComponent extends UiStatefulComponent<FooProps, FooState> {}
+        ''',
+        expectedOutput: '''
+          @Component2(isWrapper: true)
+          class FooComponent extends UiStatefulComponent2<FooProps, FooState> {}
+        ''',
+      );
+    });
+
+    test('AbstractComponent class annotation and extending class updates', () {
+      testSuggestor(
+        expectedPatchCount: 2,
+        input: '''
+          @AbstractComponent(isWrapper: true)
+          abstract class FooComponent extends UiComponent<FooProps, FooState> {}
+        ''',
+        expectedOutput: '''
+          @AbstractComponent2(isWrapper: true)
+          abstract class FooComponent extends UiComponent2<FooProps, FooState>{}
+        ''',
+      );
+    });
+
+    test(
+        'AbstractComponent class annotation and extending stateful class updates',
+        () {
+      testSuggestor(
+        expectedPatchCount: 2,
+        input: '''
+          @AbstractComponent(isWrapper: true)
+          abstract class FooComponent extends UiStatefulComponent<FooProps, FooState> {}
+        ''',
+        expectedOutput: '''
+          @AbstractComponent2(isWrapper: true)
+          abstract class FooComponent extends UiStatefulComponent2<FooProps, FooState>{}
+        ''',
+      );
+    });
+
+    test('extending class imported from react.dart updates', () {
+      testSuggestor(
+        expectedPatchCount: 2,
+        input: '''
+          import 'package:react/react.dart' as react show Component;
+          import 'package:react/react_dom.dart' as react_dom;
+        
+          class FooComponent extends react.Component {}
+        ''',
+        expectedOutput: '''
+          import 'package:react/react.dart' as react show Component, Component2;
+          import 'package:react/react_dom.dart' as react_dom;
+
+          class FooComponent extends react.Component2 {}
+        ''',
+      );
+    });
+
+    test('import from react.dart hide combinator does not update', () {
+      testSuggestor(
+        expectedPatchCount: 2,
+        input: '''
+          import 'package:react/react.dart' as react hide Component;
+          
+          @Component
+          class FooComponent extends UiComponent {}
+        ''',
+        expectedOutput: '''
+          import 'package:react/react.dart' as react hide Component;
+          
+          @Component2
```

Comment:
#nit this needs parens
```suggestion
          @Component2()
```